### PR TITLE
BUG: Mark translatable strings in Extension wizard

### PR DESCRIPTION
**Pull request by MohamedAllali**
@lassoan 
I'm not sure if the lines 361, 374 and 378 should be marked

### CODE REVIEW

File: Modules/Scripted/ExtensionWizard/ExtensionWizard.py [hunk 1]
```diff
@@ -355,7 +358,7 @@ def loadModules(path=None, depth=1, modules=None, parent=None):
                 # Add module(s) to permanent search paths, if requested
                 if dlg.addToSearchPaths:
                     settings = slicer.app.revisionUserSettings()
-                    rawSearchPaths = list(_settingsList(settings, "Modules/AdditionalPaths", convertToAbsolutePaths=True))
+                    rawSearchPaths = list(_settingsList(settings, _("Modules/AdditionalPaths"), convertToAbsolutePaths=True))
```
Comment (lassoan) [inline Modules/Scripted/ExtensionWizard/ExtensionWizard.py:361]: Should not be translated

File: Modules/Scripted/ExtensionWizard/ExtensionWizard.py [hunk 2]
```diff
@@ -368,11 +371,11 @@ def loadModules(path=None, depth=1, modules=None, parent=None):
                             modified = True
 
                     if modified:
-                        settings.setValue("Modules/AdditionalPaths", slicer.app.toSlicerHomeRelativePaths(rawSearchPaths))
+                        settings.setValue(_("Modules/AdditionalPaths"), slicer.app.toSlicerHomeRelativePaths(rawSearchPaths))
```
Comment (lassoan) [inline Modules/Scripted/ExtensionWizard/ExtensionWizard.py:374]: Should not be translated

File: Modules/Scripted/ExtensionWizard/ExtensionWizard.py [hunk 3]
```diff
@@ -368,11 +371,11 @@ def loadModules(path=None, depth=1, modules=None, parent=None):
...
 
                 # Enable developer mode (shows Reload&Test section, etc.), if requested
                 if dlg.enableDeveloperMode:
-                    qt.QSettings().setValue("Developer/DeveloperMode", "true")
+                    qt.QSettings().setValue(_("Developer/DeveloperMode"), _("true"))
```
Comment (lassoan) [inline Modules/Scripted/ExtensionWizard/ExtensionWizard.py:378]: Should not be translated

File: Modules/Scripted/ExtensionWizard/ExtensionWizard.py [hunk 4]
```diff
@@ -186,7 +189,7 @@ def setupTemplates(self):
             try:
                 self.templateManager.addPath(builtinPath)
             except:
-                qt.qWarning("failed to add built-in template path %r" % builtinPath)
+                qt.qWarning(_("failed to add built-in template path %r") % builtinPath)
```
Comment (lassoan) [inline Modules/Scripted/ExtensionWizard/ExtensionWizard.py:192]: Please use named placeholders; e.g. `{path}`

File: Modules/Scripted/ExtensionWizard/ExtensionWizard.py [hunk 5]
```diff
@@ -195,7 +198,7 @@ def setupTemplates(self):
             try:
                 self.templateManager.addPath(path)
             except:
-                qt.qWarning("failed to add template path %r" % path)
+                qt.qWarning(_("failed to add template path %r") % path)
```
Comment (lassoan) [inline Modules/Scripted/ExtensionWizard/ExtensionWizard.py:201]: Named placeholder should be used

File: Modules/Scripted/ExtensionWizard/ExtensionWizard.py [hunk 6]
```diff
@@ -240,12 +243,12 @@ def createExtension(self):
                                       " directory containing CMakeLists.txt file at '%s'" % dlg.destination)
 
                 path = self.templateManager.copyTemplate(
-                    destination, "extensions",
+                    destination, _("extensions"),
```
Comment (lassoan) [inline Modules/Scripted/ExtensionWizard/ExtensionWizard.py:246]: Don't translate

File: Modules/Scripted/ExtensionWizard/ExtensionWizard.py [hunk 7]
```diff
@@ -298,13 +301,13 @@ def selectExtension(self, path=None):
         self.extensionNameField.text = xp.project
         self.extensionLocationField.text = path
 
-        if xd.scmurl == "NA":
+        if xd.scmurl == _("NA"):
```
Comment (lassoan) [inline Modules/Scripted/ExtensionWizard/ExtensionWizard.py:304]: Do not translate

File: Modules/Scripted/ExtensionWizard/ExtensionWizard.py [hunk 8]
```diff
@@ -336,7 +339,7 @@ def loadModules(path=None, depth=1, modules=None, parent=None):
             # Get list of modules in specified path
             modules = ModuleInfo.findModules(path, depth)
         elif modules is None:
-            raise RuntimeError("loadModules require 'path' or 'modules' input")
+            raise RuntimeError("loadModules require 'path' or _('modules') input")
```
Comment (lassoan) [inline Modules/Scripted/ExtensionWizard/ExtensionWizard.py:342]: Don't translate

File: Modules/Scripted/ExtensionWizard/ExtensionWizard.py [hunk 9]
```diff
@@ -384,25 +387,25 @@ def loadModules(path=None, depth=1, modules=None, parent=None):
 
                 if len(failed):
                     if len(failed) > 1:
-                        text = "The following modules could not be registered:"
+                        text = _("The following modules could not be registered:")
                     else:
-                        text = "The '%s' module could not be registered:" % failed[0].key
+                        text = _("The '%s' module could not be registered:") % failed[0].key
```
Comment (lassoan) [inline Modules/Scripted/ExtensionWizard/ExtensionWizard.py:392]: Use named placeholders

**Review by lassoan — COMMENTED** [review]
Thank you it looks good  I just added a few trivial comments.

**Comment by MohamedAllali** [comment]
Thank you for your comment, I will correct what you pointed out.

File: Modules/Scripted/ExtensionWizard/ExtensionWizard.py (hunk 9 above)
Comment (MohamedAllali) [reply Modules/Scripted/ExtensionWizard/ExtensionWizard.py:392]: Can you explain more this?

Comment (lassoan) [reply Modules/Scripted/ExtensionWizard/ExtensionWizard.py:392]: We use placeholders everywhere where we need to insert some value into the string. For example, the code here should look something like this:

    text = _("The {moduleName} module could not be registered").format(moduleName=failed[0].key)

**Comment by lassoan** [comment]
Have you closed this intentionally?

**Comment by MohamedAllali** [comment]
No, it wasn't intentional.

Le sam. 23 mars 2024 à 15:20, Andras Lasso ***@***.***> a
écrit :

**Comment by lassoan** [comment]
It seems that it is due to that you used the same branch in several pull requests. One branch can only store one set of changes, so you must use different git branch for each pull request.

**Comment by lassoan** [comment]
We already have a pull request for the Extensions Wizard translation - https://github.com/Slicer/Slicer/pull/7698. Is this pull request still relevant?

**Pull request closed without merging**